### ISSUE TITLE
mysql cdc 使用DataStream API的时候输出的格式怎么不是通用的Json字符串呢？

### ISSUE THREAD

**Issue by liu-hai** [ververica/flink-cdc-connectors#162]
使用的列子:
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
import org.apache.flink.streaming.api.functions.source.SourceFunction;
import com.alibaba.ververica.cdc.debezium.StringDebeziumDeserializationSchema;
import com.alibaba.ververica.cdc.connectors.mysql.MySQLSource;

public class MySqlBinlogSourceExample {
  public static void main(String[] args) throws Exception {
    SourceFunction<String> sourceFunction = MySQLSource.<String>builder()
      .hostname("localhost")
      .port(3306)
      .databaseList("inventory") // monitor all tables under inventory database
      .username("flinkuser")
      .password("flinkpw")
      .deserializer(new StringDebeziumDeserializationSchema()) // converts SourceRecord to String
      .build();

    StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();

    env
      .addSource(sourceFunction)
      .print().setParallelism(1); // use parallelism 1 for sink to keep message ordering

    env.execute();
  }
}

DataStream API输出的字符串格式如下：

`SourceRecord{
	sourcePartition={server=mysql_binlog_source},
	sourceOffset={file=mysql_binary_log.000007, pos=240745028, row=1, snapshot=true}
} 
ConnectRecord{
	topic='mysql_binlog_source.flink-test.pvuv_sink',
	kafkaPartition=null, 
	key=Struct{id=HX420000_000061}, 
	keySchema=Schema{mysql_binlog_source.flink_test.pvuv_sink.Key:STRUCT}, 
	value=Struct{
		after=Struct{id=HX420000_000061,province_id=1,order_status=127001,total_amount=9999999999.889,operate_time=2021-03-27T09:07:21Z,create_time=2021-03-30T00:50:09Z,op=3},
		source=Struct{version=1.2.1.Final,connector=mysql,name=mysql_binlog_source,ts_ms=0,snapshot=true,db=flink-test,table=pvuv_sink,server_id=0,file=mysql_binary_log.000007,pos=240745028,row=0},
		op=c,
		ts_ms=1618470267061
	}, 
	valueSchema=Schema{mysql_binlog_source.flink_test.pvuv_sink.Envelope:STRUCT},
	timestamp=null, 
	headers=ConnectHeaders(headers=)
}`
这种格式想转换为json格式获取里面的值好麻烦呀，有没有什么方法呢？

**Comment by zhisheng17** [comment]
在addSource 后面加一个算子（map 或者 flatMap，转成自己想要的对象或者转成 json 格式的字符串）

**Comment by MichaelXucf** [comment]
先使用flink sql来定义cdc的表，然后再把表转换为DataStream，比如：
```java
executeSQL(tableEnv, "sql/products.sql");
Table table = tableEnv.from("products");
DataStream<Tuple2<Boolean, Products>> wordStream = tableEnv.toRetractStream(table, Products.class);
```

**Comment by MOBIN-F** [comment]
Implement DebeziumDeserializationSchema, customize the output object or format in the deserialize method, and finally output via collector.collect(T)


MySQLSource.<Rule>builder()
...
  .deserializer(new DebeziumDeserializationSchema<T>() {
                    @Override
                    public void deserialize(SourceRecord sourceRecord, Collector<T> collector) throws Exception {
                            collector.collect(T)
                    }

                    @Override
                    public TypeInformation<Rule> getProducedType() {
                        return null;
                    }
                })
                .build();

**Comment by leonardBang** [comment]
We can offer a `JsonDebeziumDeserializationSchema` to support this

**Comment by mincwang** [comment]
[common] Add debezium-json format impl of DebeziumDeserializationSchema. #344

**Comment by leonardBang** [comment]
fixed in #344
